sources/packages/api/src/services/application/application.service.ts

```diff
@@ -151,6 +320,44 @@ export class ApplicationService extends RecordDataModelService<Application> {
       .where("application.student_id = :studentId", { studentId })
       .getMany();
   }
+
+  /**
+   * Gets a application with the data needed to process
+   * the operations related to save/submitting an
+   * Student Application.
+   * @param studentId student id.
+   * @param status expected status for the application.
+   * @param [applicationId] apllication id, when possible.
+   * @returns application with the data needed to process
+   * the operations related to save/submitting.
+   */
+  private async getApplicationToSave(
+    studentId: number,
+    status: ApplicationStatus,
+    applicationId?: number,
+  ): Promise<Application> {
+    let query = this.repo
+      .createQueryBuilder("application")
+      .select([
+        "application",
+        "programYear.id",
+        "studentFiles",
+        "studentFile.uniqueFileName",
+      ])
+      .innerJoin("application.programYear", "programYear")
```

Comment:
Can you please explain, why do we have this innerjoin with programYear? if we are not using the programYear in the where clause?

Reply:
The program year is being used in the select clause and, unless we start to add the ids for relationships to the models, it will throw an exception.

Reply:
It is being used in the select clause. We have not mapped the foreign keys directly into the models, so to do a select like "programYear.id" it need be present in the FROM.

Reply:
I think, we already have the programYear.id in the application table, also it is saved as a part of the draft save and submit application. If we are planning to add the programYear.active = true condition. It makes sense